src/triage/experiments/multicore.py

```diff
@@ -56,10 +56,23 @@ def process_train_test_tasks(self, tasks):
             run_task_with_splatted_arguments, self.model_train_tester.process_task
         )
 
-        logging.info("Starting parallel testing with %s processes", self.n_processes)
-        parallelize(partial_test, tasks, self.n_processes)
+        serial_tasks, parallel_tasks = self.model_train_tester.split_tasks_by_parallelizability(tasks)
+        # some tasks we want to run serially because they implement parallelization themselves
+        # others we want to parallelize via multiprocessing
+        logging.info("Starting parallel train/testing with %s processes", self.n_processes)
+        parallelize(partial_test, parallel_tasks, self.n_processes)
         logging.info("Cleaned up concurrent pool")
 
+        for serial_task in serial_tasks:
+            final_task = serial_task
```

Comment:
is the intention to _copy_ the dictionary, because you might alter it? as is, it's unclear what purpose the rename serves.

if so, you could:

```suggestion
            final_task = copy.deepcopy(serial_task)
```

...(assuming it's a rather vanilla structure).

Or, of course, rather than dabble in new names, just redefine `serial_task` as necessary:

```
if blah:
    serial_task = copy.deepcopy(serial_task)
    ...
```